src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -1,315 +0,0 @@
-        sm_timeout = _fs().middlewared.plugins.service_monitor.socket_timeout
-
-        host_list = FreeNAS_ActiveDirectory.get_ldap_servers(host, self.config['ad_site'])
-
-        if not host_list:
-            self.alert(self.name, f'{self.name}: {host} not in connectable state. DNS query for SRV records for {host} failed.')
-            self.logger.debug(f'[ServiceMonitorThread] DNS query for SRV records for {host} failed')
-            return False
-
-        for h in host_list:
-            port_is_listening = FreeNAS_ActiveDirectory.port_is_listening(str(h.target),
-                                                                          h.port,
-                                                                          errors=[],
-                                                                          timeout=sm_timeout)
-            if port_is_listening:
-                clockskew_within_spec = self.validate_time(str(h.target), permitted_clockskew)
-                if not clockskew_within_spec:
-                    return False
-
-                return True
-            else:
-                self.logger.debug(f'[ServiceMonitorThread] Cannot connect: {h.target}:{h.port}')
-                connected = False
-
-        if not connected:
-            self.alert(self.name, f'{self.name}: Unable to contact domain controller for {host}. Domain not in connectable state.')
-
-        return connected
-
-    @private
-    def tryConnect(self, host, port):
-        if self.name == 'activedirectory':
-            domain_is_healthy = self.check_AD(host, port)
-            if not domain_is_healthy:
-                return False
-            else:
-                return True
-
-        else:
-            self.logger.debug(f'[ServiceMonitorThread] no monitoring has been written for {self.name}')
-            return False
-
-    @private
-    def getStarted(self, service):
-        max_tries = 3
-
-        for i in range(0, max_tries):
-            if service == 'activedirectory':
-                if not self.middleware.call_sync('service.started', 'cifs'):
-                    self.logger.debug("[ServiceMonitorThread] restarting Samba service")
-                    self.middleware.call_sync('service.start', 'cifs')
-            if self.middleware.call_sync('service.started', service):
-                return True
-            time.sleep(1)
-
-        return False
-
-    def run(self):
-        ntries = 0
-
-        service = self.name
-
-        while True:
-            self.finished.wait(self.frequency)
-            #
-            # We should probably have a configurable threshold for number of
-            # failures before starting or stopping the service
-            #
-            if self.finished.is_set():
-                # Thread.cancel() takes a while to propagate here
-                ServiceMonitorThread.reset_alerts(service)
-                return
-
-            if os.path.exists('/tmp/.ad_start'):
```

Comment:
So we can also drop `ServiceMonitorAlertSource` that checks for these files. But will we have a replacement?

Reply:
I was planning to rip out the service monitor. It seems redundant if there is an adequate level of alerting.